pandas/tests/groupby/test_groupby.py

```diff
@@ -2709,6 +2709,15 @@ def test_single_element_list_grouping():
     expected = [(1,), (2,)]
     assert result == expected
 
+def test_groupby_string_dtype():
+    #GH 40148
+    df = DataFrame({"str_col": ["a", "b", "c", "a"], "num_col": [1, 2, 3, 2]})
+    df["str_col"] = df['str_col'].astype("string")
+    expected = df.dtypes[0]
+    grouped = df.groupby("str_col", as_index=False)
+    avg = grouped.mean()
+    result = avg.dtypes[0]
+    tm.assert_equal(result, expected)
```

Comment:
Could you instead compare the resulting DataFrame e.g.

```
result = df.groupby("str_col", as_index=False)
expected = DataFrame(...)
tm.assert_frame_equal(result, expected)
```

Reply:
Got it!

My bad I didn't know that `assert_frame_equal` exists.
I read about it [here](https://pandas.pydata.org/docs/dev/reference/api/pandas.testing.assert_frame_equal.html?highlight=assert_frame_equal#pandas.testing.assert_frame_equal), It will compare dtypes. Sounds good 👍

Are these the required changes?
```py
expected = DataFrame({"str_col": ["a", "b", "c",], "num_col": [1.5, 2.0, 3.0]})
expected["str_col"] = expected['str_col'].astype("string")
result=grouped.mean()
tm.assert_frame_equal(result,expected)
```

Reply:
Yes

Reply:
> Yes

Thanks for confirming.